### ISSUE TITLE
Permission denied on sql log file with Docker image

### ISSUE THREAD

**Issue by madmath03** [roundcube/roundcubemail#6148]
After running docker image with Postgre (and `debug_level`= 4), I got the following error after login:
```
Warning: file_put_contents(/var/log/roundcubemail/sql.log): failed to open stream: Permission denied in /var/www/html/program/lib/Roundcube/rcube.php on line 1267
```

**Comment by J0WI** [comment]
Sadly there are still a lot of things from https://github.com/roundcube/roundcubemail/commit/4b4c4633bca167a71594d66612a26eaba193c572 that don't fit the Docker best practices. As a workaround you can use a custom value for the `ROUNDCUBEMAIL_LOG_DIR` env.

**Comment by J0WI** [comment]
@thomascube logs should be send to `stdout`/`stderr` that they are accessible via `docker logs`.

**Comment by thomascube** [comment]
> Sadly there are still a lot of things from 4b4c463 that don't fit the Docker best practices.

@J0WI Please help me out here as I'm not really a Docker expert.

**Comment by J0WI** [comment]
@thomascube some points I would change are:

Avoid extra volumes, they can be mounted anyway if someone prefer to use them.
https://github.com/roundcube/roundcubemail/blob/master/docker/Dockerfile#L24-L25

Avoid highly customized values, users can still use volume mounts.
https://github.com/roundcube/roundcubemail/blob/master/docker/docker-entrypoint.sh#L56-L57
https://github.com/roundcube/roundcubemail/blob/master/docker/docker-entrypoint.sh#L73-L74
https://github.com/roundcube/roundcubemail/blob/master/docker/docker-entrypoint.sh#L79-L82
https://github.com/roundcube/roundcubemail/blob/master/docker/docker-entrypoint.sh#L89-L92

And of course, the Docker stuff should be separated from the source and linked with Dockerhub:
https://github.com/roundcube/roundcubemail/issues/5827#issuecomment-351230892
https://github.com/roundcube/roundcubemail/issues/5827#issuecomment-358687158

But I think we already got a huge step done to fix #5827 :)

**Comment by thomascube** [comment]
@J0WI Thanks for your hints! I think the `/var/roundcube/config` volume should be kept as this is the way to extend the default config which might not be sufficient for every setup. And using a volume for the temp files is essential when running multiple instances behind a round-robin load balancer. Therefore I think these should remain as defined volumes. The `/var/www/html` volume on the other hand can be removed

**Comment by thomascube** [comment]
Logging issue fixed, more improvements will be reported to #5827.
[latest](https://hub.docker.com/r/roundcube/roundcubemail/tags/) tag updated with new build.
